src/Controller/FeedController.php

```diff
@@ -3,66 +3,96 @@
 namespace App\Controller;
 
 use App\Entity\Post;
-
 use App\Entity\Comment;
-use App\Service\PostCRUDService;
+use App\Entity\Dog;
 use App\Service\DogCRUDService;
-use App\Repository\PostRepository;
-use App\Repository\CommentRepository;
+use App\Service\CommentCRUDService;
+use App\Service\PostCRUDService;
+use App\Service\PackService;
 use App\Form\PostFormType;
 use App\Form\CommentFormType;
-use Doctrine\ORM\EntityManagerInterface;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 
 class FeedController extends AbstractController
 {
-    public function index(DogCRUDService $dogService, PostRepository $postRepository, CommentRepository $commentRepository): Response
+    public function index(
+        DogCRUDService $dogService,
+        PostCRUDService $postService,
+        CommentCRUDService $commentService, 
+        PackService $packService
+        ): Response 
     {
+
         $dogUser = $this->getUser();
-        $dogUsername = $dogUser->getUserIdentifier();
-        // dd($dogUsername);
+
         $postAuthor = $dogService->getDogNiceName($dogUser);
-        $posts = $postRepository->findAllPostsByDog($dogUser);
+
+        $posts = $dogService->getAllPosts($dogUser);
 
         if(count($posts) === 0) {
             return $this->redirectToRoute('add_post');
         }
         
-        $posts = $postRepository->findAll();
+        $myPack = $packService->getPack($dogUser);
 
-        $comments = $commentRepository->findAll();
+        $postsPack = $postService->getAllMyPackPosts($myPack);
+        
+        array_push($postsPack, $posts);
 
-        return $this->render('feed/feed.html.twig', 
-        [
-            'posts' => $posts,
-            'currentDogUser' => $postAuthor,
-            'comments' => $comments,
+        $comments = $commentService->getAllCommentsByPosts($posts);
+
+        $dogs = $dogService->getAllDogs();
+
+        $otherDogUsers = $dogService->getDogsNotInPack($myPack, $dogs);
+
+        return $this->render(
+            'feed/feed.html.twig', 
+            [
+                'posts' => $posts,
+                'currentDogUserNiceName' => $postAuthor,
+                'currentDogUser' => $dogUser,
+                'comments' => $comments,
+                'dogUsers' => $otherDogUsers,
+                'myPack' => $myPack
             ]
         );
     }
 
-    public function createComment(Request $request, EntityManagerInterface $em , Post $post) : Response
+    public function addFriend(PackService $packService, Dog $dog): Response
+    {
+        $dogUser = $this->getUser();
+        $packService->addDogToPack($dogUser, $dog);
+        return $this->redirectToRoute('feed');
+    }
+
+    public function removeFriend(PackService $packService, Dog $dog): Response
+    {
+        return $this->redirectToRoute('feed');
+    }
```

Comment:
Have a go at writing the functionality for removing a friend. I think you'll smash it out of the park. Don't forget the tests 😉

Reply:
I got this covered. Say no more 🥇 .